Src/WindowsServer/WindowsServer.Shared/AzureInstanceMetadataTelemetryInitializer.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer
+{
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.WindowsServer.Implementation;
+
+    /// <summary>
+    /// A telemetry initializer that will gather Azure Role Environment context information.
+    /// </summary>
+    public class AzureInstanceMetadataTelemetryInitializer : ITelemetryInitializer
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureInstanceMetadataTelemetryInitializer" /> class.
+        /// Creates a heartbeat property collector that obtains and inserts data from the Azure Instance
+        /// Metadata service if it is present and available to the currently running process. If it is not
+        /// present no added IMS data is added to the heartbeat.
+        /// </summary>
+        public AzureInstanceMetadataTelemetryInitializer()
+        {
+            WindowsServerEventSource.Log.TelemetryInitializerLoaded(this.GetType().FullName);
+
+            var telemetryModules = TelemetryModules.Instance;
+            foreach (var module in telemetryModules.Modules)
+
+            {
+                if (module is IHeartbeatPropertyManager)
+                {
+                    var hbeatManager = (IHeartbeatPropertyManager)module;
+
+                    // start off the heartbeat property collection process, but don't wait for it nor report
+                    // any status from here. The thread running the collection will report to the core event log.
+                    var hbeatPropertyHandler = new AzureHeartbeatProperties();
```

Comment:
variable name does not reflect class/purpose

Reply:
Fixed